test/iterators.jl

```diff
@@ -149,6 +149,30 @@ for xs in Any["abc", [1, 2, 3]]
     @test take(drop(take(xs, 3), 1), 1) === take(drop(xs, 1), 1)
 end
 
+# takewhile
+# --------
+@testset "takewhile" begin
+    @test takewhile(x -> x<4,1:10) |> collect == [1,2,3]
+    @test takewhile(x -> x<4,Iterators.countfrom(1)) |> collect == [1,2,3]
+    @test takewhile(x -> x<4,5:10) |> collect == []
+    @test takewhile(x -> true,5:10) |> collect == 5:10 |> collect
```

Comment:
```suggestion
    @test collect(takewhile(x -> true,5:10)) == 5:10
```
No need to materialize ranges for comparison.